galaxy_ng/app/access_control/mixins.py

```diff
@@ -60,15 +33,23 @@ def _set_groups(self, groups):
         if self._state.adding:
             self._groups = groups
         else:
-            current_groups = get_groups_with_perms_attached_roles(self)
+            obj = self
+
+            # If the model is a proxy model, get the original model since pulp
+            # doesn't allow us to assign permissions to proxied models.
```

Comment:
That comment is confusing to me. I know, Pulp allows to assign roles via the proxy model, but yes they need to contain permissions of that proxy model (the ones of the base model would have no effect). This was needed in order to define new permissions for the proxy model.

I believe this to be in line with: https://docs.djangoproject.com/en/3.2/topics/auth/default/#proxy-models

Reply:
@mdellweg I was getting "this role doesn't have permissions on the object" error from pulps `assign_role` function.

The role check uses `for_concrete_model=False`. https://github.com/pulp/pulpcore/blob/main/pulpcore/app/role_util.py#L34

I can add a patch for this in pulpcore if you want.

Reply:
This parameter is confusing me all the time too; `for_concrete_model=False` means the content_type fetched in the database is the content type of the proxy model [0] and not the one of the (concrete) one with the database table. That means, pulp checks permissions against the proxy model. The problem is, you cannot use "auth.view_group", but you must use the "core.view_group" permission (`pulpcore.models.Group` is the actual example i ran into trouble with). All in all, i think, this means you should use the proxy model as if it were a real one, and i think this is how django auth interprets permissions, too.

[0] https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/#django.contrib.contenttypes.models.ContentTypeManager.get_for_model

Reply:
If we got https://github.com/pulp/pulpcore/issues/2769 fixed, then we wouldn't need to proxy the models to add our helpers for getting and setting permissions.